source/game/kart/KartJump.cc

```diff
@@ -0,0 +1,258 @@
+    EGG::Vector3f vel1YCross = m_move->vel1Dir().cross(EGG::Vector3f::ey);
+    f32 vel1YCrossMag = EGG::Mathf::sqrt(vel1YCross.dot());
+    f32 pitch = EGG::Mathf::abs(EGG::Mathf::atan2(vel1YCrossMag, vel1YDot));
+    f32 angle = 90.0f - (pitch * RAD2DEG);
+    u32 weightClass = static_cast<u32>(param()->stats().weightClass);
+    f32 targetAngle = ANGLE_PROPERTIES[weightClass][static_cast<u32>(m_variant)].targetAngle;
+
+    if (angle > targetAngle) {
+        return;
+    }
+
+    f32 rotAngle = ANGLE_PROPERTIES[weightClass][static_cast<u32>(m_variant)].rotAngle;
+
+    if (angle + rotAngle > targetAngle) {
+        rotAngle = targetAngle - angle;
+    }
+
+    EGG::Matrix34f nextDir;
+    nextDir.setAxisRotation(-rotAngle * DEG2RAD, left);
+    m_move->setDir(nextDir.ps_multVector(m_move->dir()));
+    m_move->setVel1Dir(m_move->dir());
+}
+
+bool KartJump::isBoostRampEnabled() const {
+    return m_boostRampEnabled;
+}
+
+TrickType KartJump::type() const {
+    return m_type;
+}
+
+BoostRampVariant KartJump::variant() const {
+    return m_variant;
+}
+
+s16 KartJump::cooldown() const {
+    return m_cooldown;
+}
+
+void KartJump::setBoostRampEnabled(bool isSet) {
+    m_boostRampEnabled = isSet;
+}
+
+KartJumpBike::KartJumpBike(KartMove *move) : KartJump(move) {}
+
+KartJumpBike::~KartJumpBike() = default;
+
+void KartJumpBike::calcRot() {
+    /// @brief Computed using double precision, so we hard-code it.
+    constexpr f32 PI_OVER_9 = 0.34906584f;
+    /// @brief Computed using double precision, so we hard-code it.
+    constexpr f32 PI_OVER_3 = 1.0471976f;
+
+    m_angleDelta *= m_angleDeltaFactor;
+    m_angleDelta = std::max(m_angleDelta, m_properties.angleDeltaMin);
+    m_angleDeltaFactor -= m_angleDeltaFactorDec;
+    m_angleDeltaFactor = std::max(m_angleDeltaFactor, m_properties.angleDeltaFactorMin);
+    m_angle += m_angleDelta;
+    m_angle = std::min(m_angle, m_finalAngle);
+
+    switch (m_type) {
+    case TrickType::BikeFlipTrickNose:
+    case TrickType::BikeFlipTrickTail: {
+        EGG::Vector3f angles = EGG::Vector3f(-(m_angle * DEG2RAD) * m_rotSign, 0.0f, 0.0f);
+        m_rot.setRPY(angles);
+    } break;
+    case TrickType::FlipTrickYLeft:
+    case TrickType::FlipTrickYRight: {
+        EGG::Vector3f angles = EGG::Vector3f(0.0f, m_angle * DEG2RAD * m_rotSign, 0.0f);
+        m_rot.setRPY(angles);
+    } break;
+    case TrickType::BikeSideStuntTrick: {
+        f32 sin = EGG::Mathf::SinFIdx(m_angle * DEG2FIDX);
+        EGG::Vector3f angles = EGG::Vector3f(sin * -PI_OVER_9, (sin * m_rotSign) * -PI_OVER_3,
+                (sin * m_rotSign) * PI_OVER_9);
+        m_rot.setRPY(angles);
+    } break;
+    default:
+        break;
+    }
+
+    physics()->composeStuntRot(m_rot);
+}
+
+void KartJumpBike::start(const EGG::Vector3f &left) {
+    init();
+    setAngle(left);
+    state()->setInATrick(true);
+    m_cooldown = 5;
+    KartMoveBike *moveBike = reinterpret_cast<KartMoveBike *>(m_move);
```

Comment:
Is reinterpret_cast the best choice here?

Reply:
Nah it should actually be static or dynamics. We are fine to just use static imo.
I am so bad with remembering which one to use when @_@